apis/condition_set.go

```diff
@@ -221,7 +262,7 @@ func (r conditionsImpl) MarkTrue(t ConditionType) {
 	for _, cond := range r.dependents {
 		c := r.GetCondition(cond)
 		// Failed or Unknown conditions trump true conditions
-		if !c.IsTrue() {
+		if c != nil && !c.IsTrue() {
```

Comment:
this change is likely an api breaking change for knative. If the condition is nil, it means it is unknown. The expected behavior is to propagate unknown to Ready (happy)

Reply:
done